### ISSUE TITLE
Entire store HTTPS without using unsecure_base_url

### ISSUE THREAD

**Issue by mastubbs** [nexcess/magento-turpentine#675]
Hi all,

Ive been playing with and learning turpentine and varnish for a few weeks now. Its certainly a great performance improver - thank you!

Im running into a bit of a problem regarding SSL though. I want to force all pages to be Magento to be https (because of google's new ranking rules etc...). I have done this by setting unsecure_base_url to https://mysite.com (as i read others here have done also).

I am about to try and set up the reverse proxy but i see from the SSL page on the wiki that "Using HTTPS in the unsecure base URL will not work, Magento will get stuck in a redirect loop (I was not able to find a simple way around this)".

If unsecure_base_url must be http to get reverse proxy+turpentine+varnish to work then is there some other way to get entire magento store to be https? Would this be easier with nginx or pound?

Thanks very much in advance for any help or tips anyone has for this.

**Comment by aricwatson** [comment]
Closing for lack of feedback.

**Comment by maronk** [comment]
I am seeing the exact same issue of about 20% users getting a redirect loop on the frontpage.  After working on this on and off for months I decided it could be related to using varnish 3 after reading https://github.com/nexcess/magento-turpentine/pull/840 and https://github.com/nexcess/magento-turpentine/pull/844.  However I have now upgraded to varnish 4 and still seeing the same - then finally found this thread.

I have a setup with Pound, varnish on port 80 and apache on port 8008.  Note that I am bypassing varnish using Pound for specific urls as this solves a lot of problems - I don't believe this should be an issue in this case - but  I might have missed something in my thinking there.

  secure_base_url : https://mysite.com
  unsecure_base_url : http://mysite.com
  Use Secure URLs in Frontend : yes
  Offloader header : HTTP_SSL_OFFLOADED

Pound.cfg

`
ListenHTTPS
    Address x.x.x.x
    Port 443
    Cert "/etc/ssl/certs/xyz.pem"
    xHTTP 2
    RewriteLocation 1
    Ciphers "ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS"
    Disable SSLv3
    AddHeader "Ssl-Offloaded: 1"
    HeadRemove "X-Forwarded-Proto"
    AddHeader "X-Forwarded-Proto: https"
    Service
        URL "/(mysecretadminpath.*|checkout.*|customer.*)"
        BackEnd
            Address 127.0.0.1
            Port    8008
        End
    End
    Service
        BackEnd
            Address 127.0.0.1
            Port    80
        End
    End
End
`

Any ideas what could be wrong?
